transcom/mymove
# B-20313: Prevent Role List Duplication

## [B-20313](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20313)

## Summary

In Admin App when viewing an Office Users Roles the roles list would sometimes duplicate. It is very inconsistent when this happens and may not even happen for us locally. I remember during PI planning that this was being cause because the GET Office-users call would return duplicate roles. Although i could not reproduce locally I was able to work with Alex to reproduce in loadtest to make sure that what was happening with the API endpoint during PI planning is exactly what is causing the duplication on the Admin UI now. (See screenshot below)

Notice the roles are duplicated in the payload. The AdminAPI displays what it receives from this call.
![RoleDuplicationIssue](https://github.com/user-attachments/assets/f84eb955-57ad-4cb1-a381-84bb9ead43f3)


[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the Admin App
2. Login as an admin user
3. Go to view an Office user
4. Make sure roles are not duplicating in the user roles list
![image](https://github.com/user-attachments/assets/572d051e-9f75-4792-b4aa-0ec0a826a8c9)


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
